### PR TITLE
Catch invalid filter: tcpdump()

### LINKED ISSUE #2172: Exception raised with bad BPF filter
The following command raises an exception:
```
sniff(offline=IP()/UDP(), filter="bad filter", quiet=True)
```

The `tcpdump()` must detect the error from the `tcpdump` command `(tcpdump: syntax error in filter expression: syntax error`).

### PR DESCRIPTION
**Pull request by gpotter2**
An alternative to #2205. To discuss

fix #2205 
fix #2172

### CODE REVIEW

**Comment by guedou** [comment]
I tried it and I think that it is a good compromise to fix the issue. However, I think that sniff should not throw the exception in interactive mode but display a "critical" message.

**Comment by gpotter2** [comment]
We want that issue to end the `sniff()` loop so we'll need an exception to raise. I'm not sure we want to create another special one for that.

**Comment by guedou** [comment]
Another approach to fix #2205 could be to use `compile_flter()` in `AsyncSniffer.run()` to check the BFP filter. What do you think?

**Comment by gpotter2** [comment]
That's a good idea. tcpdump requires libpcap anyways

File: scapy/utils.py [hunk 1]
```diff
@@ -971,6 +971,10 @@ def __call__(cls, filename):
         """
         i = cls.__new__(cls, cls.__name__, cls.__bases__, cls.__dict__)
         filename, fdesc, magic = cls.open(filename)
+        if not magic:
+            raise Scapy_Exception(
+                "The file descriptor is closed !"
```
Comment (p-l-) [inline scapy/utils.py:976]: Something like that maybe?
```suggestion
                "No data could be read!"
```

**Comment by gpotter2** [comment]
We use `pcap_compile_nopcap` if possible, which does not require privileges. `pcap_open_live` is only used when we can't guess the interface linktype.

**Comment by guedou** [comment]
@gpotter2 I thought that it was a regression regarding v2.4.3. It is not.

However, I think that it will be good to be able to use `sniff(offline=IP()/UDP(), filter="ip")` without being root.

**Comment by gpotter2** [comment]
That works on my Ubuntu machine without root. What are you using ?

**Comment by guedou** [comment]
macOS 10.14.6

**Pull request merged**